src/main/kotlin/net/emdal/tank/Graph.kt

```diff
@@ -1,20 +1,30 @@
 package net.emdal.tank
 
-data class Graph(val query: String = "", val aliases: List<Pair<String, Entity>> = emptyList()) {
+data class Graph(val query: String = "", val aliases: List<Pair<String, Entity?>> = emptyList()) {
   fun <T : Node> node(
     alias: String = "",
     node: T,
-    block: T.() -> String = { "" }
+    block: T.() -> String? = { null }
   ): Graph = node(node, alias, block)
 
   fun <T : Node> node(
     node: T,
     alias: String = "",
-    block: T.() -> String = { "" }
+    block: T.() -> String? = { null }
   ): Graph {
     return this.copy(
-      query = query + "($alias:${node.label} {${node.block()}})",
+      query = query + "($alias:${node.label}${node.block()?.let { " { $it}" } ?: ""})",
```

Comment:
Can we make this an extension function `?.let { " { $it}" } ?: ""`?

Reply:
yes!